### PR TITLE
'Group you joined' list

### LINKED ISSUE #225: Groups you joined: group list
## Motivations
Have a list of the groups the user joined. Inspired by https://france-ioi.github.io/algorea-designs/01.Groups_02.Composition_l.Belong-to-groups

## Subtasks
* On the "group you joined" page
* Add a subtile "Groups you belong to"
* Add a table with all groups from the membershipsView service (already used for the nav menu)
* As column: "Name", "Type", "Joined on"   
* "Date joined" is date+time and sortable
* No gear widget, no selection box in front

### PR DESCRIPTION
**Pull request by GabrielVidal1**
Fixes #225

### CODE REVIEW

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      <td>{{ group.group.name }}</td>
+      <td>{{ group.group.type }}</td>
+      <td>{{ group.memberSince | date:'short' }}</td>
+    </tr>
+  </ng-template>
+
+  <ng-template pTemplate="emptymessage" let-columns>
+    <tr>
+      <td [attr.colspan]="columns.length">
+        <p class="empty-message">This list is empty. Check below the different ways to join a group or sub-groups.</p>
```
Comment (smadbe) [inline src/app/modules/group/components/joined-group-list/joined-group-list.component.html:33]: While there is nothing to check below, no need to write to check below.

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.scss [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+:host {
+  position: relative;
+  display: block;
+
+  ::ng-deep .ui-table {
+    margin-bottom: 1rem;
+    font-family: 'titillium', sans-serif;
+
+
+    .ui-table-thead > tr > th {
```
Comment (smadbe) [inline src/app/modules/group/components/joined-group-list/joined-group-list.component.scss:26]: When the ngprime10 will be merge, you 'll need to rebase and convert all ui-table-* to p-datatable-*

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.scss [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      white-space: nowrap;
+      font-size: 1.2rem;
+      padding : 0 1rem;
+    }
+
+    .ui-table-tbody > tr:last-child > td {
+      border-bottom: none;
+    }
+
+    .ui-table-tbody > tr.ui-state-highlight > td {
```
Comment (smadbe) [inline src/app/modules/group/components/joined-group-list/joined-group-list.component.scss:77]: ui-state-highlight -> p-highlight

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.scss [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      color: #fff;
+    }
+
+  }
+}
+
+
+
+
+.show-all {
```
Comment (smadbe) [inline src/app/modules/group/components/joined-group-list/joined-group-list.component.scss:87]: nothing should be out of the `:host` I think in scss files of components.

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  styleUrls: [ './joined-group-list.component.scss' ]
+})
+export class JoinedGroupListComponent implements OnDestroy, OnInit {
+
+  state: 'error' | 'ready' | 'fetching' = 'fetching';
+  currentSort: string[] = [];
+
+  data: JoinedGroup[] = [];
+
+  private dataFetching = new Subject<{ sort: string[] }>();
```
Comment (smadbe) [inline src/app/modules/group/components/joined-group-list/joined-group-list.component.ts:20]: Other similar components use `reloadData` functions and `unsubscribe`. I am not sure it was worth changing this here... but ok, leave it.

File: src/app/modules/group/pages/joined-groups/joined-groups.component.html [hunk 6]
```diff
@@ -4,3 +4,9 @@
     <span class="group-subtitle">{{ subtitle }}</span>
   </div>
 </div>
+
```
Comment (smadbe) [inline src/app/modules/group/pages/joined-groups/joined-groups.component.html:7]: Ok, please fix (in another PR ?) the subtitle. Actually put the title and subtitle in template, I find it confusing to have these in variable anyway. And actually the font does not seem really correct either.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        )
+      )
+    ).subscribe(
+      state => {
+        this.state = state.tag;
+        if (isReady(state)) this.data = state.data;
+      },
+      _err => {
+        this.state = 'error';
+      }
```
[the remark this answers is not in the document]
Comment (GabrielVidal1) [reply src/app/modules/group/components/joined-group-list/joined-group-list.component.ts]: fixed

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.scss (hunk 4 above)
Comment (GabrielVidal1) [reply src/app/modules/group/components/joined-group-list/joined-group-list.component.scss]: fixed, I removed this part, it was not used anyway

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.scss (hunk 3 above)
Comment (GabrielVidal1) [reply src/app/modules/group/components/joined-group-list/joined-group-list.component.scss]: fixed

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.scss (hunk 2 above)
Comment (GabrielVidal1) [reply src/app/modules/group/components/joined-group-list/joined-group-list.component.scss]: fixed

File: src/app/modules/group/components/joined-group-list/joined-group-list.component.html (hunk 1 above)
Comment (GabrielVidal1) [reply src/app/modules/group/components/joined-group-list/joined-group-list.component.html]: fixed

File: src/app/core/http-services/joined-groups.service.ts [hunk 8]
```diff
@@ -1,23 +1,29 @@
...
-  name: string
+interface RawJoinedGroup{
+  action: 'invitation_accepted' | 'join_request_accepted' | 'joined_by_code' | 'added_directly',
+  group: {
+    description?: string,
+    id: string,
+    name: string,
+    type: 'Class' | 'Team' | 'Club' | 'Friends' | 'Other' | 'Base',
+  },
+  member_since: string,
```
[the remark this answers is not in the document]
Comment (GabrielVidal1) [reply src/app/core/http-services/joined-groups.service.ts]: fixed

File: src/app/core/http-services/joined-groups.service.ts [hunk 9]
```diff
@@ -1,23 +1,29 @@
...
 import { map } from 'rxjs/operators';
 import { appConfig } from 'src/app/shared/helpers/config';
 
-export interface Group {
-  id: string,
-  name: string
+interface RawJoinedGroup{
+  action: 'invitation_accepted' | 'join_request_accepted' | 'joined_by_code' | 'added_directly',
+  group: {
+    description?: string,
```
[the remark this answers is not in the document]
Comment (GabrielVidal1) [reply src/app/core/http-services/joined-groups.service.ts]: fixed

File: src/app/core/components/group-nav/group-nav.component.ts [hunk 10]
```diff
@@ -29,7 +29,9 @@ export class GroupNavComponent {
   onTabOpen(event: {index: number}): void {
     this.focusOnGroupNav.emit();
     const service = event.index == joinGroupTabIdx ?
```
[the remark this answers is not in the document]
Comment (GabrielVidal1) [reply src/app/core/components/group-nav/group-nav.component.ts]: fixed

File: src/app/modules/group/pages/joined-groups/joined-groups.component.html (hunk 6 above)
Comment (GabrielVidal1) [reply src/app/modules/group/pages/joined-groups/joined-groups.component.html:7]: fixed

File: src/app/modules/group/pages/joined-groups/joined-groups.component.scss [hunk 11]
```diff
@@ -23,3 +23,7 @@
   .group-info-content {
   }
 }
+
+.bg-white {
+  background-color: white;
+}
```
[the remark this answers is not in the document]
Comment (GabrielVidal1) [reply src/app/modules/group/pages/joined-groups/joined-groups.component.scss]: fixed

File: src/app/modules/group/pages/joined-groups/joined-groups.component.html [hunk 12]
```diff
@@ -1,7 +1,7 @@
 <div class="group-info-container">
   <div class="group-info-header">
-    <span class="group-title">{{ title }}</span>
-    <span class="group-subtitle">{{ subtitle }}</span>
+    <span class="group-title">Joined Groups</span>
+    <span class="group-subtitle">Here are the groups you joined, you can leave them ore add new ones</span>
```
Comment (smadbe) [inline src/app/modules/group/pages/joined-groups/joined-groups.component.html:4]: typo "ore"

**Review by smadbe — APPROVED** [review]

**Pull request merged**